app/models/user.rb

```diff
@@ -81,6 +83,14 @@ def problem_severity(category)
     problem_settings[category.to_s]&.to_sym || Problem::DEFAULT_SEVERITIES[category.to_sym]
   end
 
+  def self.from_omniauth(auth)
+    find_or_create_by(auth_provider: auth.provider, auth_uid: auth.uid) do |user|
+      user.email = auth.info.email
+      user.password = user.password_confirmation = Devise.friendly_token[0, 20]
```

Comment:
I am not sure about the security implications of this approach.

The token issued by the provider is base64-encoded JSON, so it has a lot of identical bytes at the beginning…

Users created from OIDC should not have a password set and should not be able to login with a password.

Reply:
Thanks, I'll check that, this is straight from their example code. I may need to alter the validations a bit to remove the password requirement in OIDC situations.

Reply:
Hmm, if Omniauth say this is a good idea, I'd actually trust it ;).